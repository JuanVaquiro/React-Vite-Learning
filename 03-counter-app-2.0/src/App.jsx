import PropTypes from 'prop-types';

const newMessage = {
  message: "Hola Padre este es el msm;",
  conter: "Saludo Dirigido",
};

const FntSuma = (b, c) => {
  const a = b * c;
  return a;
};

const App = ({ title, subTitle, name }) => {

  return (
    <div className="App">
      <header className="App-header">
        <h1>{title}</h1>
        <h2>{name}</h2>
        <p>{subTitle }</p>    
        <span>{FntSuma(2, 3)}</span> 
        <br/>
        <br/>
        <code>{JSON.stringify(newMessage)}</code>
      </header>
    </div>
  );
}

App.propTypes = {
  title: PropTypes.string.isRequired,
  subTitle: PropTypes.number,
}

App.defaultProps = {
  // title: 'quihubo',
  subTitle: 2,
  name: 'JUAN D'
}


export default App;
