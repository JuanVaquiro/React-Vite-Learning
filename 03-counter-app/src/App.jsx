// Funcional Component 
import React, { Fragment } from 'react';
import PropTypes from 'prop-types';
// imr

const PrimeraApp = ( { saludo, subtitulo } ) => {

  return (
    <Fragment>  
      <h1>{ saludo }</h1>
      <p>{ subtitulo }</p>
    </Fragment>
  );
};

PrimeraApp.propTypes = {
  saludo: PropTypes.string.isRequired
}

PrimeraApp.defaultProps = {
  subtitulo: 'MI primera App'
}

export default PrimeraApp;  
