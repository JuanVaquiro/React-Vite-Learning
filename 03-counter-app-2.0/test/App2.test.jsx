import { render, screen } from "@testing-library/react";
import App from "../src/App";

describe('Pruebas en <App2 />', () => {

    // Mejorar el codigo de testing en el componente <App />

    const title = 'Hola, soy Juan'
    const subTitle = 'el subtitulo'

    test('debe de hacer match con el Snapshot ', () => {

        const { container } = render( <App title={ title } /> )
        expect( container ).toMatchSnapshot();
    });

    test('debe de mostrar le mensaje: "Hola, soy Juan"', () => {
        
        render( <App title={ title } /> )

        // "screen" el objeto que se renderiza
        // screen.debug() : renderiza el DOM del compenete y ver el objeto
        expect( screen.getByText( title ) ).toBeTruthy();

    });

    test('debe de mostrar el titulo en un <h1>', () => {
        
        render( <App title={ title } /> )

        expect( screen.getByRole("heading", { level: 1 }).innerHTML ).toContain( title );
    });

    test('debe de mostrar el subtitulo enviado por props', () => {
        
        render( 
            <App 
                title={ title }
                subTitle={ subTitle } 
            /> 
        )

        expect( screen.getAllByText(subTitle).length ).toBe(2);
    });
    
});