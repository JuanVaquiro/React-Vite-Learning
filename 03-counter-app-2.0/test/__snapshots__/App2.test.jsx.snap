// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <App2 /> debe de hacer match con el Snapshot  1`] = `
<div>
  <div
    class="App"
  >
    <header
      class="App-header"
    >
      <h1
        data-testid="test-title"
      >
        Hola, soy Juan
      </h1>
      <h2>
        JUAN D
      </h2>
      <p>
        este es el subtitulo
      </p>
      <p>
        este es el subtitulo
      </p>
      <span>
        6
      </span>
      <br />
      <br />
      <code>
        {"message":"Hola Padre este es el msm;","conter":"Saludo Dirigido"}
      </code>
    </header>
  </div>
</div>
`;
